feature/issue-198: update the on the same modal of create and add server action and also have added the update query

Usamavirk444 opened the pull request.

Usamavirk444 committed:
feature/issue-198: update the on the same modal of create and add server action and also have added the update query

usama-tariq1 commented on src/components/Dashboard/Projects/CreateNewProject.tsx:
```diff
@@ -41,53 +41,98 @@ const channelSchema = z.object({
   project_type: z.boolean().optional()
 })
 
+type ProjectFormData = z.infer<typeof channelSchema>;
```
please rename schema to project schema

usama-tariq1 commented on src/components/Dashboard/Projects/CreateNewProject.tsx:
```diff
@@ -41,53 +41,98 @@ const channelSchema = z.object({
-
-function CreateNewProject() {
-
+function CreateNewProject({ defaultValues, isEditing = false }: {
```
this modal is not just create modal please name it accordingly

usama-tariq1 commented on src/components/Dashboard/Projects/CreateNewProject.tsx:
```diff
@@ -41,53 +41,98 @@ const channelSchema = z.object({
   const [isOpen, setIsOpen] = useState(false)
   const [projects, setProjects] = useAtom(projectStore.projects)
-  const [createProjectLoading, createProjectData, createProjectError, createProject] = useServerAction(CreateProjectAction)
+  const [createProjectLoading, , , createProject] = useServerAction(CreateProjectAction)
```
please try npm run build to see if leaving the prams empty gives any errors

usama-tariq1 commented on src/components/Dashboard/Projects/CreateNewProject.tsx:
```diff
@@ -41,53 +41,98 @@ const channelSchema = z.object({
+  }, [searchParams])
 
   useEffect(() => {
+    if (!isClient) return
```
what is isClient used for ?

usama-tariq1 commented on src/components/Dashboard/Projects/CreateNewProject.tsx:
```diff
@@ -41,53 +41,98 @@ const channelSchema = z.object({
 
   useEffect(() => {
+    if (!isClient) return
     GetSpaceBySlugAction(spaceSlug || "", channelSlug || "").then((currentSpace) => {
```
space slug and channelslug could be null and result in 500

usama-tariq1 commented on src/db/data-access/project-management/query.ts:
```diff
@@ -31,4 +31,22 @@ export async function getProjectById(projectId: string) {
\ No newline at end of file
+}
+
+export async function UpdateProject(project_data: InsertProject) {
```
function starts with lower case

usama-tariq1 replied:
please receive partial type

usama-tariq1 commented on src/components/Dashboard/Projects/ProjectCards.tsx:
```diff
@@ -77,6 +78,13 @@ function ProjectCards({ project }: Props) {
             View Details
           </LinkAsButton>
+          
+          <LinkAsButton asChild>
```
i believe theres no need for link as button component and modal to be initiated in each card 
project modal should only render once in whole page

usama-tariq1 replied:
please set the state that can be listened in modal to update accordingly

usama-tariq1 commented on src/server-actions/ProjectManagement/projectManagement.ts:
```diff
@@ -17,6 +17,15 @@ export const CreateProjectAction = CreateServerAction(true, async (project_data:
 
 )
 
+export const UpdateProjectAction = CreateServerAction(true, async (project_data: InsertProject) => {
```
receive partial type

Usamavirk444 committed:
feature/issue-198: rename schema to project schema
create modal please name it accordingly
could be null and result in 500
button component and modal to be initiated in each card
project modal should only render once in whole page
function starts with lower case
please receive partial type
receive partial type

Commit:
fix(issue-198): hydration fix
